Sources/Utility/Process.swift

```diff
@@ -165,6 +172,24 @@ public final class Process: ObjectIdentifierProtocol {
         self.redirectOutput = redirectOutput 
     }
 
+    /// Validate that the program provided exists on the search path and is executable.
+    internal func validateArgumentIsExecutable(_ program: String) throws {
+
+        // check to see if the argument is already listed in the vaildated cache
+        guard !Process.validatedExecutables.contains(program) else {
+            return
+        }
+        let envSearchPaths = Utility.getEnvSearchPaths(pathString: getenv("PATH"), currentWorkingDirectory: currentWorkingDirectory)
+        guard let executableLocation = Utility.lookupExecutablePath(filename: program, searchPaths: envSearchPaths) else {
+            throw Process.Error.missingProgram(program: program)
+        }
+        guard localFileSystem.isExecutableFile(executableLocation) else {
```

Comment:
if you mean `lookupExecutablePath`, then unfortunately no - it doesn't. It's poorly named, but I've left the name in place (perhaps I should update it?) - that method only iterates through the search paths to find the executable, it doesn't actually *verify* it's executable as well.

I could combine these together and update that function, and return a single error (missing or not executable), but with them separate I can be more specific about the error. I honestly wasn't sure which was better, so i aimed for more information to debug/understand the problem if one happens.